### ISSUE TITLE
crash on trying to browse repository from spyder

### ISSUE THREAD

**Issue by varadpatil** [spyder-ide/spyder#15544]
### What steps will reproduce the problem?

1. create new conda python environment
1. install spyder 5.0.1
1. open new terminal and install gitk
1. in spyder project pane, rightclick a file and select 'Browse repository'

### Traceback
```python-traceback
Traceback (most recent call last):
  File "/home/varad/miniconda3/envs/atl/lib/python3.9/site-packages/spyder/plugins/explorer/widgets/explorer.py", line 409, in <lambda>
    triggered=lambda: self.vcs_command('browse'),
  File "/home/varad/miniconda3/envs/atl/lib/python3.9/site-packages/spyder/plugins/explorer/widgets/explorer.py", line 1448, in vcs_command
    vcs.run_vcs_tool(path, action)
  File "/home/varad/miniconda3/envs/atl/lib/python3.9/site-packages/spyder/utils/vcs.py", line 87, in run_vcs_tool
    programs.run_program(tool, args, cwd=path)
  File "/home/varad/miniconda3/envs/atl/lib/python3.9/site-packages/spyder/utils/programs.py", line 278, in run_program
    return subprocess.Popen(fullcmd, **subprocess_kwargs)
  File "/home/varad/miniconda3/envs/atl/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/varad/miniconda3/envs/atl/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
NotADirectoryError: [Errno 20] Not a directory: '/home/varad/CWD/algo_trading/dev/utils/README.md'
```

## Versions

* Spyder version: 5.0.1 
* Python version: 3.9.4
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Linux 5.8.0-50-generic

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0               :  7.23.1 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.7.3 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=1.0                  :  2.7.2 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2           :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.0 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.0.3             :  5.1.0 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.8.3                 :  0.9.7 (OK)
setuptools >=39.0.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  3.5.3 (OK)
spyder_kernels >=2.0.1;<2.1.0 :  2.0.2 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  1.0.2 (OK)
xdg >=0.26                    :  0.26 (OK)
zmq >=17                      :  20.0.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=2.0.0            :  None (NOK)
numpy >=1.7                   :  None (NOK)
pandas >=1.1.1                :  None (NOK)
scipy >=0.17.0                :  None (NOK)
sympy >=0.7.3                 :  None (NOK)
```

**Comment by juanis2112** [comment]
Hi @varadpatil thanks for reporting this. I can successfully reproduce this issue. We will fix it in our next version.
This seems related to #15458

**Comment by ccordoba12** [comment]
> This seems related to #15458

I think it's basically the same problem and it should be fixed in our next version (5.1.0), to be released at the end of the month.
